### PR TITLE
feat: Allow 'false' as a proper value for credentialManagerOverride APIs

### PR DESCRIPTION
**Pull request by zFernand0**
**What It Does**

Allows credential manager override to specify a value of `false` 

**How to Test**

- Download this VSIX: https://ent.box.com/s/5t9a90ewumbalqoj75too7vgyeehix7l
- Disable the `zowe.security.secureCredentialsEnabled` property
- Remove the secure properties from the `secure[]` array.
- Attempt to store credentials (`right-click > Manage Profile > Update Credentials`)
- You should see a warning about credentials being stored in plain-text
  image

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

This is related to the following Zowe Explorer issue:
- https://github.com/zowe/zowe-explorer-vscode/issues/2622

### CODE REVIEW

**Review by anaxceron — APPROVED** [review]
Great!

**Comment by adam-wolfe** [comment]
Not really related to this PR, but it might be good to update the string "Saving credentials will update the local information file.". It's not really clear what "local information file" means.

https://github.com/zowe/zowe-explorer-vscode/blob/61992f1ff951911774d5c06c46598c24751065d8/packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts#L322

**Review by t1m0thyj — DISMISSED** [review]
Looks pretty good, thanks @zFernand0! I may reach out later for more details about how to test this PR 😋

File: packages/imperative/src/security/src/CredentialManagerFactory.ts [hunk 1]
```diff
@@ -82,13 +82,13 @@ export class CredentialManagerFactory {
...
-        const displayName = params.displayName == null ? params.service : params.displayName;
+        const displayName = params.displayName ?? params.service;
 
         // If a manager override was not passed, use the default keytar manager
-        const Manager = params.Manager == null ? DefaultCredentialManager : params.Manager;
+        const Manager = params.Manager ?? DefaultCredentialManager;
 
         // Default invalid on failure to false if not specified
-        params.invalidOnFailure = params.invalidOnFailure == null ? false : params.invalidOnFailure;
+        params.invalidOnFailure = params.invalidOnFailure ?? false;
```
[the remark this answers is not in the document]
Comment (traeok) [reply packages/imperative/src/security/src/CredentialManagerFactory.ts:91]: Just noticed that line 91 can leverage nullish coalescing assignment `??=`:

```typescript
params.invalidOnFailure ??= false;
```

**Review by traeok — APPROVED** [review]

**Review by traeok — APPROVED** [review]
Thanks Fernando for the enhancement! Tested locally and LGTM 😋

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @zFernand0 for the VSIX 😋

**Pull request merged**